compute/instances/creating-instance-with-custom-machine-type/custom_machine_type_helper.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snippets
+
+// [START compute_custom_machine_type_helper_class]
+import (
+	"fmt"
+	"strings"
+)
+
+const (
+	N1        = "custom"
+	N2        = "n2-custom"
+	N2D       = "n2d-custom"
+	E2        = "e2-custom"
+	E2_MICRO  = "e2-custom-micro"
+	E2_SMALL  = "e2-custom-small"
+	E2_MEDIUM = "e2-custom-medium"
+)
+
+type TypeLimit struct {
+	allowedCores     []int
+	minMemPerCore    int
+	maxMemPerCore    int
+	allowExtraMemory bool
+	extraMemoryLimit int
+}
+
+var (
+	CPUSeries_E2_Limit        = TypeLimit{MakeRange(2, 33, 2), 512, 8192, false, 0}
```

Comment:
These inline structs seem a little large. I suggest adding the field names as well, for clarity.

Reply:
Ditto on the constant names.